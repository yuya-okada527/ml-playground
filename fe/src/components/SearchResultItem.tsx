/* eslint-disable jsx-a11y/anchor-is-valid */
// TODO 暫定対応
import {
  createStyles,
  Grid,
  ListItem,
  makeStyles,
  Paper,
  Theme,
  Typography,
} from "@material-ui/core";
import Link from "next/link";
import Image from "material-ui-image";
import React from "react";
import { Movie } from "../interfaces";
import { makeTitle } from "./MovieDetail";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    movieCardTitle: {
      margin: theme.spacing(1),
    },
    link: {
      textDecoration: "none",
    },
    searchResultItem: {
      paddingLeft: 0,
      paddingBottom: theme.spacing(1),
      paddingTop: theme.spacing(0),
    },
    searchResultItemPaper: {
      width: "100%",
      transition: "0.2s",
      "&:hover": {
        backgroundColor: "#eee",
        opacity: 0.8,
      },
    },
  })
);

type SearchResultItemProps = {
  movie: Movie;
};

/**
 * 検索結果を描画する
 */
const SearchResultItem: React.FC<SearchResultItemProps> = ({ movie }) => {
  const classes = useStyles();
  return (
    <Link href={`/movies/${movie.movie_id}`}>
      <a className={classes.link}>
        <ListItem className={classes.searchResultItem} key={movie.movie_id}>
          <Paper
            className={classes.searchResultItemPaper}
            variant="outlined"
            square
          >
            <Grid container>
              <Grid item xs={1}>
                <Image src={movie.poster_url} aspectRatio={9 / 12} />
              </Grid>
              <Grid item xs={11}>
                <Typography className={classes.movieCardTitle} variant="h6">
                  {makeTitle(movie)}
                </Typography>
              </Grid>
            </Grid>
          </Paper>
        </ListItem>
      </a>
    </Link>
  );
};

export default SearchResultItem;
