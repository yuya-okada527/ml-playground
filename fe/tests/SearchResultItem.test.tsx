import React from "react";
import { shallow } from "enzyme";
import SearchResultItem from "../src/components/SearchResultItem";
import { emptyMovie } from "./TestUtil";

describe("SearchResultItem", () => {
  it("Snapshot Test", () => {
    // テストデータ
    const movie = emptyMovie();
    movie.movie_id = 0;
    movie.japanese_title = "タイトル";
    movie.poster_url = "http://movie.co.jp/image.png";
    movie.release_year = 2000;
    const searchResultItem = shallow(<SearchResultItem movie={movie} />);

    // 検証
    expect(searchResultItem).toMatchSnapshot();
  });
});
