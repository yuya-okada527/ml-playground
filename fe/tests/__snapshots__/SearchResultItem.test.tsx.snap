// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultItem Snapshot Test 1`] = `
<Link
  href="/movies/0"
  key="0"
>
  <a
    className="makeStyles-link-2"
  >
    <WithStyles(ForwardRef(ListItem))
      className="makeStyles-searchResultItem-3"
      key="0"
    >
      <WithStyles(ForwardRef(Paper))
        className="makeStyles-searchResultItemPaper-4"
        square={true}
        variant="outlined"
      >
        <WithStyles(ForwardRef(Grid))
          container={true}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={1}
          >
            <Image
              animationDuration={3000}
              aspectRatio={0.75}
              color="#fff"
              disableError={false}
              disableSpinner={false}
              disableTransition={false}
              errorIcon={
                <Memo
                  style={
                    Object {
                      "color": "#e0e0e0",
                      "height": 48,
                      "width": 48,
                    }
                  }
                />
              }
              loading={
                <WithStyles(ForwardRef(CircularProgress))
                  size={48}
                />
              }
              src="http://movie.co.jp/image.png"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={11}
          >
            <WithStyles(ForwardRef(Typography))
              className="makeStyles-movieCardTitle-1"
              variant="h6"
            >
              タイトル
               (2000) 
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Paper))>
    </WithStyles(ForwardRef(ListItem))>
  </a>
</Link>
`;
